src/Drawer/index.js

```diff
@@ -31,10 +32,21 @@ const Drawer = ({
   const navRef = useRef(null);
 
   const isTransitioning = useMountTransition(isOpen, 300);
+  const { m } = useBreakpoints();
   const isHorizontal = position === "bottom" || position === "top";
+  const isVerticalMobileDrawer = (!m) && !isHorizontal;
+
   // The depth is how far the drawer opens out, but the CSS prop depends
   // on whether the Drawer is vertical or not
-  const depthStyle = isHorizontal ? { height: depth } : { width: depth };
+  const depthStyle = {
+    width: isHorizontal ? "auto" : depth,
+    height: isHorizontal ? depth : "auto",
+  };
+  if (isVerticalMobileDrawer) {
+    depthStyle.width = "100%";
+    depthStyle.height = "100%";
```

Comment:
needed to also set height to be 100% here - otherwise it was only occupying a small amount of the vertical space